### PR TITLE
load data: move param and some init logic into importer pkg

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #40499 

Problem Summary:

### What is changed and how it works?
- move some common code in `load_data.go` into `importer` pkg to reuse in physical import
- this pr is big most due to **move errors.go into a separate package to avoid import cycle**

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/importer/import.go [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+	}
+
+	e.insertColumns = reorderedColumns
+
+	return nil
+}
+
+// GetInsertColumns get column list need to insert into target table.
+// this list include all columns and in the same order as in fieldMappings and ColumnAssignments
+func (e *LoadDataController) GetInsertColumns() []*table.Column {
```
Comment (lance6716) [inline executor/importer/import.go:511]: maybe make this member public

**Review by lance6716 — APPROVED** [review]

**Comment by lance6716** [comment]
/cc @hawkingrei 

This PR moves the error definitions from executor to executor/exeerrors

File: executor/exeerrors/errors.go [hunk 2]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package executor
```
Comment (hawkingrei) [inline executor/exeerrors/errors.go:15]: In fact, We have some errors defined at ```util/dbterror``` such as DDL.

Comment (D3Hunter) [reply executor/exeerrors/errors.go]: https://github.com/pingcap/tidb/pull/42075/commits/80a4048477469869714db3f2ec5645bd0e114641

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**